automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/OrchestrationApiSpec.scala

```diff
@@ -1,109 +1,94 @@
+    with Matchers
+    with ScalaFutures
+    with Eventually {
+
   implicit override val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(5, Seconds)))
+  val billingAccountName: String = ServiceTestConfig.Projects.billingAccountId
 
   def nowPrint(text: String): Unit = {
     println(s"${Instant.now} : $text")
   }
 
   "Orchestration" - {
     "should link an eRA Commons account with access to the TARGET closed-access dataset" in {
-      withCleanUp {
-        val user = UserPool.chooseAuthDomainUser
-        implicit val userToken: AuthToken = user.makeAuthToken()
-
-        register cleanUp resetNihLinkToInactive
+      val user = UserPool.chooseAuthDomainUser
+      implicit val userToken: AuthToken = user.makeAuthToken()
 
-        Orchestration.NIH.addUserInNIH(OrchConfig.Users.targetJsonWebTokenKey)
-
-        verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", false), NihDatasetPermission("TARGET", true)))
-      }
+      Orchestration.NIH.addUserInNIH(OrchConfig.Users.targetJsonWebTokenKey)
+      try verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", false), NihDatasetPermission("TARGET", true)))
+      finally resetNihLinkToInactive
     }
 
     "should link an eRA Commons account with access to the TCGA closed-access dataset" in {
-      withCleanUp {
-        val user = UserPool.chooseAuthDomainUser
-        implicit val userToken: AuthToken = user.makeAuthToken()
+      val user = UserPool.chooseAuthDomainUser
+      implicit val userToken: AuthToken = user.makeAuthToken()
 
-        register cleanUp resetNihLinkToInactive
-
-        Orchestration.NIH.addUserInNIH(OrchConfig.Users.tcgaJsonWebTokenKey)
-
-        verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", true), NihDatasetPermission("TARGET", false)))
-      }
+      Orchestration.NIH.addUserInNIH(OrchConfig.Users.tcgaJsonWebTokenKey)
+      try verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", true), NihDatasetPermission("TARGET", false)))
+      finally resetNihLinkToInactive
     }
 
-    "should link an eRA Commons account with access to none of the supported closed-access datasets" in {
-      withCleanUp {
-        val user = UserPool.chooseAuthDomainUser
-        implicit val userToken: AuthToken = user.makeAuthToken()
-
-        register cleanUp resetNihLinkToInactive
+  "should link an eRA Commons account with access to none of the supported closed-access datasets" in {
+    val user = UserPool.chooseAuthDomainUser
+    implicit val userToken: AuthToken = user.makeAuthToken()
 
-        Orchestration.NIH.addUserInNIH(OrchConfig.Users.genericJsonWebTokenKey)
+    Orchestration.NIH.addUserInNIH(OrchConfig.Users.genericJsonWebTokenKey)
 
-        verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", false), NihDatasetPermission("TARGET", false)))
-      }
-    }
+    try verifyDatasetPermissions(Set(NihDatasetPermission("TCGA", false), NihDatasetPermission("TARGET", false)))
+    finally resetNihLinkToInactive
```

Comment:
just out of curiosity, did you opt for try...finally over withCleanUp because it is more idiomatic to scala? (asking to better understand the pros/cons)

Reply:
I am also totally fine with the change and had the same question.

Reply:
intellij complains about inheriting from the `CleanUp` trait. It also uses a self-type that bamboozles me. I guess I've removed it in places were it's simple to add a try/finally as part of my jihad on inheritance.